cmd/system-probe/modules/network_tracer.go

```diff
@@ -378,3 +383,57 @@ func writeConntrackTable(table *tracer.DebugConntrackTable, w http.ResponseWrite
 		w.WriteHeader(500)
 	}
 }
+
+// traceLogConnections logs detailed connection information to help investigate NPM/USM
+// customer issues by providing visibility into network connections and their metadata.
+func traceLogConnections(id string, cs *network.Connections) {
```

Comment:
One small suggestion as enabling trace logs is going to spam system-probe - 
Either add a prefix to the logs, or maybe change the feature enabling the trace logs, as you might not be able to see only your trace logs when needed
I do generally recommend (unless you've already done that) to simulate the scenario in which system-probe is enabled with trace logs, and ensure we can easily obtain the information we need